ARCHITECTURE.md

```diff
@@ -0,0 +1,54 @@
+# Architecture
+
+This document describes the high-level architecture of SkyPortal.
+If you want to familiarize yourself with the code base, you are just in the right place!
+
+See also the [feature guide](./doc/adding_features.md), which walks through specifics of the code base, as well as the [developer guide](./doc/dev.md), which has notes on code styling and testing.
+
+## Bird's Eye View
+
+SkyPortal is an example of a Target and Observation Manager (TOM), which , are designed to support the astronomical community to efficiently work with astronomical data sets, identify sources of interest, and obtain follow-up. It aims to be a sophisticated ``full-stack transient ecosystem'' that integrates capabilities of telescope scheduling, observing optimization, spatial catalogs, astronomical data analysis, and others into a single software stack.
+
+## Front-end
+
+The SkyPortal front-end uses React, a component-based UI library, in conjunction with Redux, a front-end state-management library.
+
+Each component definition typically goes in its own file in `static/js/components`.
+
+We bundle all of our Redux-related code (action types, action creators, reducer) associated with a particular branch of the application state together in a file in the `static/js/ducks` directory ([read more about "ducks" modules here](https://github.com/erikras/ducks-modular-redux)).
```

Comment:
So that it doesn't look like pure magic, I would mention that we hydrate new tabs using the redux state of previous tabs to minimize redundant interactions with the server.

Reply:
added this line.